backend/src/Edna/ExperimentReader/Error.hs

```diff
@@ -4,24 +4,38 @@ module Edna.ExperimentReader.Error
 
 import Universum
 
-import Codec.Xlsx (ParseError)
+import Codec.Xlsx (CellValue, ParseError)
+import Data.Typeable (cast)
 import Fmt (Buildable(..), (+|), (|+))
 
+import Edna.ExperimentReader.Types (CellType, PointYX(..))
+
 data ExperimentParsingError
-  = UnexpectedCellType
-  | InvalidCell (Int, Int)
+  = forall a . Typeable a => UnexpectedCellType PointYX (CellType a) (Maybe CellValue)
+  | InvalidCell PointYX
   | EmptyCell
   | WorksheetNotFound
   | TabletStartNotFound
   | FileParsingError ParseError
   | NoConcentrationTablet
-  deriving stock (Show, Generic, Eq)
+
+deriving stock instance Show ExperimentParsingError
+instance Eq ExperimentParsingError where
+  UnexpectedCellType a b c == UnexpectedCellType d e f = a == d && c == f && Just b == (cast e)
+  InvalidCell a == InvalidCell b = a == b
+  EmptyCell == EmptyCell = True
+  WorksheetNotFound == WorksheetNotFound = True
+  FileParsingError a == FileParsingError b = a == b
+  NoConcentrationTablet == NoConcentrationTablet = True
+  _ == _ = False
```

Comment:
There is a chance that someone will add a new error constructor and forget to write `NewConstr == NewConstr = True` here. To avoid that, we can write `2 * n` cases instead of `n + 1`:
```
FileParsingError a == FileParsingError b = a == b
FileParsingError _ == _ = False
NoConcentrationTablet == NoConcentrationTablet = True
NoConcentrationTablet == _ = False
…
```

I propose to do it, it's not still linear number of cases, not quadratic, so shouldn't be a big problem.